common/src/model_deployment/deploy.py

```diff
@@ -0,0 +1,243 @@
+        type=str,
+        help="Json file to which deployment details will be written",
+    )
+    # parse args
+    args = parser.parse_args()
+    print("args received ", args)
+    # return args
+
+    # Validating passed input values
+    if args.max_concurrent_requests_per_instance < 1:
+        parser.error("Arg max_concurrent_requests_per_instance cannot be less than 1")
+    if args.request_timeout_ms < 1 or args.request_timeout_ms > 90000:
+        parser.error("Arg request_timeout_ms should lie between 1 and 90000")
+    if args.max_queue_wait_ms < 1 or args.max_queue_wait_ms > 500:
+        parser.error("Arg max_queue_wait_ms should lie between 1 and 500")
+
+
+def get_endpoint(args):
+    """Return online or batch endpoint."""
+    endpoint = ManagedOnlineEndpoint(name=args.endpoint_name, auth_mode="key")
+    print("Endpoint created with name ", endpoint.name)
+    return endpoint
+
+
+def get_ml_client():
+    """Return ML client."""
+    msi_client_id = os.environ.get("DEFAULT_IDENTITY_CLIENT_ID")
+    credential = ManagedIdentityCredential(client_id=msi_client_id)
+
+    run = Run.get_context(allow_offline=False)
+    ws = run.experiment.workspace
+
+    ml_client = MLClient(
+        credential=credential,
+        subscription_id=ws._subscription_id,
+        resource_group_name=ws._resource_group,
+        workspace_name=ws._workspace_name,
+    )
+    return ml_client
+
+
+def main(args):
+    """Run main function."""
+    ml_client = get_ml_client()
+    model_id = (Path(args.registration_details)).read_text()
+    endpoint = get_endpoint(args)
+
+    ml_client.begin_create_or_update(endpoint).wait()
+
+    # deployment
+    deployment = ManagedOnlineDeployment(
+        name=args.deployment_name,
+        endpoint_name=endpoint.name,
+        model=model_id,
+        instance_type=args.instance_type,
+        instance_count=args.instance_count,
+        request_settings=OnlineRequestSettings(
+            max_concurrent_requests_per_instance=args.max_concurrent_requests_per_instance,
+            request_timeout_ms=args.request_timeout_ms,
+            max_queue_wait_ms=args.max_queue_wait_ms,
+        ),
+        liveness_probe=ProbeSettings(
+            failure_threshold=args.failure_threshold_liveness_probe,
+            timeout=args.timeout_liveness_probe,
+            period=args.period_liveness_probe,
+            initial_delay=args.initial_delay_liveness_probe,
+        ),
+        readiness_probe=ProbeSettings(
+            failure_threshold=args.failure_threshold_readiness_probe,
+            success_threshold=args.success_threshold_readiness_probe,
+            timeout=args.timeout_readiness_probe,
+            period=args.period_readiness_probe,
+            initial_delay=args.initial_delay_readiness_probe,
+        ),
+        egress_public_network_access = args.egress_public_network_access,
+    )
+    ml_client.online_deployments.begin_create_or_update(deployment).wait()
```

Comment:
Should we catch exception in case endpoint or deployment creation fails?

Reply:
Thanks . Added